src/navigation/onboardingStateNavigation.tsx

```diff
@@ -0,0 +1,135 @@
+import Debug from 'debug';
+import React, {ReactNode} from 'react';
+import {State} from 'xstate';
+import {APP_ID} from '../@config/constants';
+import {translate} from '../localization/Localization';
+import {OnboardingContext, OnboardingMachine} from '../machines/onboardingMachine';
+import store from '../store';
+import {ScreenRoutesEnum} from '../types';
+import {
+  IOnboardingMachineContext,
+  IOnboardingPersonalData,
+  OnboardingEvents,
+  OnboardingEventTypes,
+  OnboardingInterpretType,
+  OnboardingStates,
+} from '../types/onboarding';
+import {LOGIN_SUCCESS} from '../types/store/user.action.types';
+import RootNavigation from './rootNavigation';
+
+const debug: Debug.Debugger = Debug(`${APP_ID}:storageService`);
+export const onboardingStateNavigationListener = (
+  onboardingMachine: OnboardingInterpretType,
+  state: State<
+    IOnboardingMachineContext,
+    OnboardingEventTypes,
+    any,
+    {
+      value: any;
+      context: IOnboardingMachineContext;
+    },
+    any
+  >,
+  navigation?: any,
+) => {
+  const onBack = () => onboardingMachine.send(OnboardingEvents.PREVIOUS);
+  const onNext = () => onboardingMachine.send(OnboardingEvents.NEXT);
+  const context = onboardingMachine.getSnapshot().context;
+
+  const nav = navigation ?? RootNavigation;
+  if (nav === undefined || !nav.isReady()) {
+    debug(`navigation not ready yet`);
+    return;
+  }
+  if (state.matches(OnboardingStates.welcomeIntro)) {
```

Comment:
Nah will leave it as is, because the matches function does a few things that are not too heavyweight, and prevent us from potentially introducing bugs later on if we extend the machine. If else if vs case statements are syntactic sugar. I do like case statements better when there there are multiple else-ifs, but in this case (pun intended) there are no constants, and thus I cannot use it.